contracts/src/LiquidityPoolV2.sol

```diff
@@ -102,47 +146,60 @@ contract LiquidityPoolV2 is ReentrancyGuardUpgradeable, AccessManagedUpgradeable
         // Transfer the input tokens from the user to the contract
         ERC20Upgradeable(tokenIn).transferFrom(msg.sender, address(this), amountIn);
 
-        // Calculate the output amount using the constant product formula
-        uint256 amountInWithFee = amountIn * (10000 - platformFee) / 10000;
+        // Calculate the output amount using the constant product formula, adjusted for the platform fee
+        uint256 amountInWithFee = (amountIn * (10000 - platformFee)) / 10000;
```

Comment:
**adjustForDecimals** (see last comment of the file to understand)

Replace amountIn in the transfers but the adjustedAmountIn